datafusion/sqllogictest/test_files/joins.slt

```diff
@@ -3673,7 +3673,7 @@ physical_plan
 11)------------ProjectionExec: expr=[1 as c, 3 as d]
 12)--------------PlaceholderRowExec
 
-query IIII
+query IIII rowsort
```

Comment:
I found these to be needed due to unstable ordering: https://github.com/apache/arrow-datafusion/actions/runs/8722139931/job/23927618449?pr=10114

Reply:
I think we fixed them on main too with  https://github.com/apache/arrow-datafusion/pull/10122 and https://github.com/apache/arrow-datafusion/pull/10120